laravel-app/app/Services/SearchService.php

```diff
@@ -4,15 +4,36 @@
 
 use App\Interfaces\SearchServiceInterface;
 use App\Models\User;
-use Illuminate\Contracts\Database\Eloquent\Builder;
+use App\Models\UserPost;
+use Illuminate\Database\Eloquent\Builder as EloquentBuilder;
+use Illuminate\Pagination\LengthAwarePaginator;
 
 class SearchService implements SearchServiceInterface
 {
     /**
-     * To search the User table using the Eloquent of matching results.
+     * To search the User and Posts of followed users for the mathching keyword.
      */
-    public function searchUser(String $query): Builder
+    public function searchUser(string $query): LengthAwarePaginator
     {
-        return User::where('username', 'like', '%' . $query . '%');
+        /** @var User $authUser */
+        $authUser = auth()->user();
+
+        $users = User::where('username', 'like', '%' . $query . '%')
+            ->whereIn('is_verified', [1])
+            ->get()->all();
+
+        $postsQuery = UserPost::whereIn('user_id', $authUser->following->pluck('id')->push($authUser->id))
+            ->where(function (EloquentBuilder $queryBuilder) use ($query) {
+                $queryBuilder->where('content', 'regexp', '\b' . $query . '\b');
+            })->get()->all();
```

Comment:
Same here.

Reply:
I did that initially sir, but since I have to use an array when combining the query result for user and posts I had to add the all() to convert the collection to array.
![image](https://github.com/bpocrafael/microblog-project/assets/144191038/8daf2103-7d0b-4de5-9443-690511e873c7)


I tried using concat to merge the 2 collections and they work, however I get an error from phpstan like this
![image](https://github.com/bpocrafael/microblog-project/assets/144191038/fdab99c9-db97-4442-b25f-46931b2fb19f)